ops/ansible/roles/bfd-server/templates/bfd-server.sh.j2

```diff
@@ -95,6 +95,8 @@ check_endpoint() {
 # Runs test queries to verify that the server is working as expected. If not, issues a 'kill -INT' on the server's process to shut it down.
 ##
 stop_service_if_failing() {
+  # Reject all external traffic to port 7443 until the application has started successfully
+  sudo iptables -A INPUT -p tcp -i eth0 --dport 7443 -j REJECT
```

Comment:
Specifying the interface is not necessary (and interface names can and have changed depending on the whim of the distro), so i'd drop the -i bit.

Reply:
That is a _little_ verbose, but I thought it was added for good reason. I was under the impression that an underspecified rule **without** `--in-insert` _or_ `-i` would apply to all interfaces, including `lo`. My evaluation of the strategy seemed to support that, but I'm always trying to simplify, so I thought I'd double check:

The underspecified `iptables` rule appears to be too aggressive:
```
# demonstrate successful connection attempt
$ nc -zv localhost 7443
Connection to localhost (127.0.0.1) 7443 port [tcp/oracleas-https] succeeded!

# introduce rejection rule for all traffic on 7443
$ iptables -A INPUT -p tcp --dport 7443 -j REJECT

# demonstrate undesirable refused connection attempt
$ nc -zv localhost 7443
nc: connect to localhost (127.0.0.1) port 7443 (tcp) failed: Connection refused

# remove rejection rule
$ iptables -D INPUT -p tcp --dport 7443 -j REJECT
```

The more specific rule works as expected:
```
# demonstrate successful connection attempt
$ nc -zv localhost 7443
Connection to localhost (127.0.0.1) 7443 port [tcp/oracleas-https] succeeded!

# introduce rejection rule for all traffic on 7443 across eth0 interface
$ iptables -A INPUT -p tcp -i eth0 --dport 7443 -j REJECT

# demonstrate another successful connection attempt
$ nc -zv localhost 7443
Connection to localhost (127.0.0.1) 7443 port [tcp/oracleas-https] succeeded!

$ iptables -D INPUT -p tcp -i eth0 --dport 7443 -j REJECT
# remove rejection rule
```

Maybe the happy medium is being explicit about what we don't want to block instead, i.e.

`iptables -A INPUT -p tcp ! -i lo --dport 7443 -j REJECT`

This seems to work. What do you think?

Reply:
Yeah, normally I have DROP for default policies and my rules _allow_ traffic, not the other way around. So yeah, that last rule you suggested should work fine.